### DISCUSSION TITLE
Individual changes and building of firmware

### DISCUSSION THREAD

**Discussion by Ford32** [moonglow/FlashForge_Marlin#153] [Q&A] [answered]
Hi, 

I already managed to upload a .bin file to my flashforge dreamer.
Now i would like do do some modifications like silent stepper drivers and maybe the heating cartridge of thermistor.

1) what .zip should di i need to change? FlashForge_Marlin-marlin_2.0.x.zip or FlashForge_Marlin-0.15.1.zip?
2) what software do i need for the changes and the building? 
  -> Is Arduino IDE sufficient to do the changes and get my .bin file? (if yes, what board do i need to select to build it?)
  
  - also the windows installer did not work for me because it was missing some .dll file
 
Many Thanks to all the contributors of this project! 

I did struggle quite a bit because many things that are obvious for you guys, weren't for me it seems.

Greetings.

**Comment by zerog2k** [comment, marked as answer]
>what .zip should di i need to change?

check out the code base with `git clone ...` or if you just want to checkout a snapshot, and modify that, just click on "code", then download .zip snapshot of this branch:

![image](https://user-images.githubusercontent.com/3943170/235220842-0940c097-014c-4209-b2e2-4321d81e4f53.png)

>what software do i need for the changes and the building?

https://platformio.org/install/cli

[marlin_builder.sh](https://github.com/moonglow/FlashForge_Marlin/blob/marlin_2.0.x/marlin_builder.sh) is the script that sits on above the customized marlin directory to help build the fw.

I would advise you that if you are not familiar with Marlin modification, compilation, and flashing/reflashing, you may want to consider this very advanced territory.

**Comment by Ford32** [comment]
Thank You!
